dashboard/app/models/user.rb

```diff
@@ -2261,6 +2261,41 @@ def providers
     end
   end
 
+  # Returns number of times a user has attempted to join a section in the last 24 hours
+  # Returns nil if no section join attempts
+  def num_section_attempts
+    properties['section_attempts'] || 0
+  end
+
+  # There aare two possible states in which we would want to reset section attempts
+  # 1) Initialize for the first time 2) 24 hours have passed since last attempt
+  def reset_section_attempts?
+    !num_section_attempts || !properties['section_attempts_last_reset'] || num_section_attempts == 0 || (DateTime.now - DateTime.parse(properties['section_attempts_last_reset'])).to_i > 0
```

Comment:
Why do we need to reset section attempts if it nil/0?  (Also, depending on your answer to my other question above, we may not need to check for nil here.)

Reply:
Agreed completely. Given `num_section_attempts` will always return 0 if the value is `nil`, we don't need to check for `nil` values here.